packages/flame/lib/src/camera/behaviors/follow_behavior.dart

```diff
@@ -59,19 +62,17 @@ class FollowBehavior extends Component {
 
   @override
   void update(double dt) {
-    final delta = target.position - owner.position;
-    if (horizontalOnly) {
-      delta.y = 0;
-    }
-    if (verticalOnly) {
-      delta.x = 0;
-    }
-    final distance = delta.length;
+    _tempDelta.setValues(
+      verticalOnly ? 0 : target.position.x - owner.position.x,
+      horizontalOnly ? 0 : target.position.y - owner.position.y,
+    );
+
+    final distance = _tempDelta.length;
     if (distance > _speed * dt) {
-      delta.scale(_speed * dt / distance);
+      _tempDelta.scale(_speed * dt / distance);
     }
-    if (delta.x != 0 || delta.y != 0) {
-      owner.position = delta..add(owner.position);
+    if (_tempDelta.x != 0 || _tempDelta.y != 0) {
+      owner.position = _tempDelta..add(owner.position);
```

Comment:
to go all the way avoiding object creation entirely, this could be owner.position.setValues(owner.position.x + delta.x, ...)

sadly we lose all the easiness of using vectorial math at all when doing things like this

Reply:
> this could be owner.position.setValues(owner.position.x + delta.x, ...)

I thought the same. But unfortunately doing so causes some of the tests to fail, because the `.setValues` bypasses the `position` setter of `PositionProvider`. Some components do extra stuff in that setter.

Reply:
oof, seems like a footgun waiting to happen :(